src/firebaseConfig.ts

```diff
@@ -5,74 +5,121 @@
 // 'npm run generate:json-schema' to regenerate the schema files.
 //
 
-export type DatabaseConfig =
+// base configs
+type DeployAsset = {
+  predeploy?: string | string[];
+  postdeploy?: string | string[];
+};
+
+type DatabaseOne = {
+  rules: string;
+} & DeployAsset;
+
+type DatabaseMany = ({
+  rules: string;
+} & (
   | {
-      rules?: string;
+      instance: string;
+      target?: string;
     }
   | {
-      target?: string;
       instance?: string;
-      rules: string;
-    }[];
+      target: string;
+    }
+) &
+  DeployAsset)[];
 
-export type FirestoreConfig = {
-  rules?: string;
-  indexes?: string;
+type HostingRedirects = ({ source: string } | { regex: string }) & {
+  destination: string;
+  type: number;
 };
 
-export type FunctionsConfig = {
-  // TODO: Add types for "backend" and "runtime"
-  source?: string;
-  ignore?: string[];
-  predeploy?: string[];
+type HostingRewrites = ({ source: string } | { regex: string }) &
+  (
+    | { destination: string }
+    | { function: string }
+    | {
+        run: {
+          serviceId: string;
+          region?: string;
```

Comment:
I am not sure about this value.

The [Hosting Cloud Functions docs state](https://firebase.google.com/docs/hosting/functions)
>Important: Firebase Hosting supports Cloud Functions in us-central1 only.

The [Hosting Cloud Run docs](https://firebase.google.com/docs/hosting/cloud-run) do not make a clarification, just that `us-central1` is the default

The [full-config docs](https://firebase.google.com/docs/hosting/full-config#rewrite-cloud-run-container) shows the entire list of available Cloud Run regions. Though when I tested this in February I couldn't get Hosting rewrites to work in non-`us-central1` regions.

Needless to say I am confused of the requirements here. I feel we should use string literals for this with all valid region options.

Reply:
Must have messed something up during my last testing because this seems to work now.